pinot-broker/src/main/java/org/apache/pinot/broker/routing/RoutingTable.java

```diff
@@ -0,0 +1,44 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.broker.routing;
+
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+import org.apache.pinot.core.transport.ServerInstance;
+
+
+public class RoutingTable {
+  private Map<ServerInstance, List<String>> _serverInstanceToSegmentsMap;
+  private List<String> _segmentsWithNoReplicas;
+
+  public RoutingTable(@Nullable Map<ServerInstance, List<String>> serverInstanceToSegmentsMap,
```

Comment:
nit: Both arguments are nullable

Reply:
I don't want `segmentsWithNoReplicas` to be nullable. Being null could indicate a lot of meanings, and it's hard to handle. Here `segmentsWithNoReplicas` should only be either empty or non-empty. That's the only information we want.

Reply:
agreed